pallets/xdns/src/lib.rs

```diff
@@ -403,20 +409,29 @@ pub mod pallet {
         fn add_new_token(
             token_id: [u8; 4],
             gateway_id: [u8; 4],
-            token_props: TokenSysProps,
+            token_props: TokenInfo,
         ) -> DispatchResult {
             // early exit if record already exists in storage
             if <Tokens<T>>::contains_key(token_id) {
```

Comment:
how do we not know that this collides with pallet-assets/asset-registry? How would we use it?

Reply:
This is used for whitelisting assets for SFX invocations for specific gateways. I honestly don't really know how this should tie into pallet-assets.

Reply:
Pallet assets has a metadata field here, so we'd ensure that the token id exists for this. Otherwise we will have collisions when we come to use the assets registered here, and have to double the lookups when using assets already onboarded in the system.

For these gateway assets, we'd just register them with 0 ED, mint no tokens, the owner is sudo & the metadata is just this.